tools/src/main/scala/scala/scalanative/build/Config.scala

```diff
@@ -68,6 +71,9 @@ sealed trait Config {
   /** Create a new config with given compilation options. */
   def withCompileOptions(value: Seq[String]): Config
 
+  /** Create a new config with given target architecture. */
+  def withTargetArchitecture(value: TargetArchitecture): Config
+
```

Comment:
Do we really need to have both `target triple` and `target architecture` as two separate concepts in the public API? It seems like `TargetArchitecture` is derivative from the triple and we can only have triple without changing the build API.